documentation/modules/ROOT/partials/modules/mysql-connector/con-mysql-connector-events.adoc

```diff
@@ -3,15 +3,67 @@
 [id="mysql-connector-events_{context}"]
 = MySQL connector events
 
-All data change events produced by the {prodname} MySQL connector contain a key and a value. The change event key and the change event value each contain a _schema_ and a _payload_ where the schema describes the structure of the payload and the payload contains the data.
+The {prodname} MySQL connector generates a data change event for each row-level `INSERT`, `UPDATE`, and `DELETE` operation. Each event contains a key and a value. The structure of the key and the value depends on the table that was changed. 
 
-WARNING: The MySQL connector ensures that all Kafka Connect schema names adhere to the link:http://avro.apache.org/docs/current/spec.html#names[Avro schema name format]. This is important as any character that is not a latin letter or underscore is replaced by an underscore which can lead to unexpected conflicts in schema names when the logical server names, database names, and table names container other characters that are replaced with these underscores.
+{prodname} and Kafka Connect are designed around _continuous streams of event messages_. However, the structure of these events may change over time, which can be difficult for consumers to handle. To address this, each event contains the schema for its content, which makes each event self-contained. The following skeleton JSON shows the basic structure of a change event: 
 
-== Change event key
+[source,json,index=0]
+----
+{
+ "schema": { //<1>
+   ...
+  },
+ "payload": { //<2>
+   ...
+ },
+ "schema": { //<3> 
+   ...
+ },
+ "payload": { //<4>
+   ...
+ },
+}
+----
+
+.Overview of change event basic content
+[cols="1,2,7",options="header"]
+|===
+|Item |Field name |Description
+
+|1
+|`schema`
+|The first `schema` field is part of the event key. It specifies a Kafka Connect schema that describes what is in the event key's `payload` portion. In other words, the first `schema` field describes the structure of the primary key for the table that was changed.
```

Comment:
Could also be a unique key if the table doesn't have a PK. Plus, a user can explicitly override the key column(s). Not sure how to frame this in terms of the default that applies in 99% of cases (PK) and the 1% of cases where this actually could look differently.

Reply:
I changed the last sentence in the first `schema` description to this: 

In other words, the first `schema` field describes the structure of the primary key, or the unique key if the table does not have a primary key, for the table that was changed. 

After that, I'd like to add a sentence but I am not sure how to word it. I'm thinking that "override the key column(s)" is part of converter configuration, but maybe I'm wrong about that. I think it would be helpful to say what you might override the key column with.  I would add something like this, but I expect this is not quite right and hope you will fix it: 

Rarely, converter configuration overrides the key column with some other structure. In this case, the first `schema` field describes that structure.

Reply:
@gunnarmorling - wdyt about this update?

Reply:
For the latest sentence, its not a configuration that comes from the converter but actually the connector itself, see the `message.key.columns` configuration option.  This is the explicit override config setup Gunnar was describing.

Reply:
@Naros How's this: 

It is possible to override the table's primary key by setting the {link-prefix}:{link-mysql-connector}#mysql-property-message-key-columns[`message.key.columns` connector configuration property]. In this case, the first schema field describes the structure of the the key identified by that property.